scripts/tools/EXT_Exp_MetricProgrammable.rst

```diff
@@ -0,0 +1,120 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+    t=tags['$t']
+    T=t.upper()
+%>
+:orphan:
+
+.. _ZET_experimental_programmable_metric:
+
+==========================================
+Metric Programmable Experimental Extension
+==========================================
+
+API
+----
+* Enumerations
+
+    * ${t}_metric_programmable_param_type_exp_t
+
+* Structures
+
+    * ${t}_metric_programmable_exp_properties_t
+    * ${t}_metric_programmable_param_exp_desc_t
+    * ${t}_metric_programmable_param_value_exp_t
+    * ${t}_metric_programmable_param_value_exp_desc_t
+    * ${t}_value_exp_desc_t
+    * ${t}_value_uint64_range_exp_t
+
+* Functions
+
+    * ${t}MetricProgrammableGetExp
+    * ${t}MetricProgrammableGetPropertiesExp
+    * ${t}MetricProgrammableGetParamDescriptionExp
+    * ${t}MetricProgrammableGetParamValueDescriptionExp
```

Comment:
generally `desc` are input args, so here it seems we are providing desc to gather output right? is there different way to name this API/ output structures ?

Reply:
Let me check this.
But please share if there is a guideline about the names.

Reply:
renamed to info